src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = {};
+
+  Object.assign(currentState, state);
+
+  for (let i = 0; i < actions.length; i++) {
+    let updatedState = {};
+
+    Object.assign(updatedState, currentState);
+
+    if (actions[i].type === 'addProperties') {
+      Object.assign(updatedState, actions[i].extraData);
+    } else if (actions[i].type === 'removeProperties') {
+      actions[i].keysToRemove.forEach((key) => delete updatedState[key]);
+    } else if (actions[i].type === 'clear') {
+      updatedState = {};
```

Comment:
When the 'clear' action is executed, `updatedState` is set to an empty object. However, this does not clear the `currentState` object, which may lead to incorrect state history. Consider using `Object.keys(updatedState).forEach(key => delete updatedState[key]);` to clear the object while maintaining the reference.